util/predicates/cluster_predicates.go

```diff
@@ -188,3 +188,27 @@ func ClusterUnpausedAndInfrastructureReady(logger logr.Logger) predicate.Funcs {
 	// Use any to ensure we process either create or update events we care about
 	return Any(log, createPredicates, updatePredicates)
 }
+
+// ClusterIsManagedTopology returns a Predicate that returns true when cluster.Spec.Topology
+// is NOT nil and false otherwise.
+func ClusterIsManagedTopology(logger logr.Logger) predicate.Funcs {
```

Comment:
I'm not sure about the name. Maybe `ClusterHasManagedTopology`, `ClusterIsTopologyManaged` or `ClusterIsUsingClusterClass` would be slightly better? (just throwing a few alternatives out there)

Reply:
ClusterTopologyManaged is closer to clusterTopologyOwned (the label name)

Reply:
`ClusterIsTopologyOwned` would be a candidate too :)

Reply:
Let's wait a bit for further opinions, as it's a publicly exposed func.